### ISSUE TITLE
Build documentation for each pull request

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#602]
### Description

When making changes that involve documentation (e.g. new features), it's helpful to be able to accurately preview that documentation as it will be shown on read the docs. This is theoretically possible by talking to the RTD people, as described in https://blog.readthedocs.com/building-docs-for-pull-requests/

### Done Checklist

- [ ] Produced code for required functionality
- [ ] Tests written and coverage checked
- [ ] Code review performed
- [ ] Documentation on Google Docs  (if applicable)
- [ ] Documentation in repo
- [ ] Version number reflects new status
- [ ] CHANGELOG.md updated
- [ ] Team demo

**Comment by huonw** [comment]
I've emailed them to ask for access to the beta feature.

**Comment by huonw** [comment]
They've enabled it for us. I've switched our readthedocs web hook (https://github.com/stellargraph/stellargraph/settings/hooks/50326049) to respond to PR pushes too:

![image](https://user-images.githubusercontent.com/1203825/72027586-bc7cb880-32d3-11ea-8350-0c8ba6652c35.png)

It's correctly picking them up and starting builds, but those builds are failing when installing dependencies, e.g. https://readthedocs.org/projects/stellargraph/builds/10225724/ is a build for https://github.com/stellargraph/stellargraph/pull/628:

```
/home/docs/checkouts/readthedocs.org/user_builds/stellargraph/envs/628/bin/python -m pip install --exists-action=w --cache-dir /home/docs/checkouts/readthedocs.org/user_builds/stellargraph/.cache/pip -r requirements.txt
Collecting networkx>=2.2
  Using cached https://files.pythonhosted.org/packages/41/8f/dd6a8e85946def36e4f2c69c84219af0fa5e832b018c970e92f2ad337e45/networkx-2.4-py3-none-any.whl
Requirement already satisfied: setuptools>=39.1.0 in /home/docs/checkouts/readthedocs.org/user_builds/stellargraph/envs/628/lib/python3.6/site-packages (from -r requirements.txt (line 2)) (41.0.1)
Collecting tensorflow>=2.0.0
  Downloading https://files.pythonhosted.org/packages/85/d4/c0cd1057b331bc38b65478302114194bd8e1b9c2bbc06e300935c0e93d90/tensorflow-2.1.0-cp36-cp36m-manylinux2010_x86_64.whl (421.8MB)
Killed


Command killed due to excessive memory consumption
```

**Comment by huonw** [comment]
The error above was fixed in #631, and we got someone to sign into zenhub and add the app to the StellarGraph org to be able to report commit statuses:

![image](https://user-images.githubusercontent.com/1203825/72565655-70023000-3906-11ea-97bb-250c9c3a9475.png)
